### PR TITLE
Adding Contributing doc

### PR DESCRIPTION
**Pull request by williamcaban**
## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] Optimization
- [x] Documentation Update

## Description

Adding CONTRIBUTING.md

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,9 +1,14 @@
+# ignore python and editors cache
 __pycache__
+venv
+.env*
+.vscode
+.DS_Store
+
+# ignore locally significat directories
```
Comment (bparees) [inline .gitignore:8]:
```suggestion
# ignore locally significant directories
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+# Install dependencies
+pip install -r requirements.txt
+```
+
+Happy hacking!
+
+## Before a PR
+
+If updating `requirements.txt` identify if doing so for the `main` branch or a release branch and follow the corresponding guidance. If a dependency is no longer required, remove it from the list.
```
Comment (bparees) [inline CONTRIBUTING.md:38]: i don't understand this part.  If you're updating requirements.txt for main branch, the PR will be against the main branch.

if you're doing it for a release branch (note: we're probably about to delete our release branches for now as they don't make sense right now) you'd submit the PR against the release branch.

so you don't need to "identity" which you're doing it for

**Review by bparees — COMMENTED** [review]
couple comments and also if you can squash this down to a single commit to keep our history a bit cleaner it'd be appreciated (we can also squash on merge but then we get all the concatenated commit messages)

File: src/logger.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import logging, sys, os
```
Comment (bparees) [inline src/logger.py:1]: pretty sure you didn't mean to include this file in this PR

Comment (williamcaban) [reply src/logger.py:1]: yeah, this is not part of it. When I try to create the other PR it updated this one

File: CONTRIBUTING.md (hunk 2 above)
Comment (williamcaban) [reply CONTRIBUTING.md:38]: The intent is to have the guidance for whoever cut the releases.

Comment (bparees) [reply CONTRIBUTING.md:38]: ok then maybe put this in a separate section about cutting releases

**Pull request closed without merging**